skyportal/handlers/api/observation_plan.py

```diff
@@ -1252,20 +1252,28 @@ def observation_simsurvey(
         injection_filename: str
             Path to file for injestion as a simsurvey.models.AngularTimeSeriesSource
     . Defaults to nsns_nph1.0e+06_mejdyn0.020_mejwind0.130_phi30.txt from https://github.com/mbulla/kilonova_models.
-        Returns
-        -------
-        dict
-            success : bool
-                Whether the request was successful or not, returning
-                a sensible error in 'reason'
-            name : str
-                suggested filename based on `source_name` and `output_format`
-            data : str
-                binary encoded data for the plot (to be streamed)
-            reason : str
-                If not successful, a reason is returned.
     """
 
+    localization = (
+        DBSession()
+        .execute(sa.select(Localization).where(Localization.id == localization_id))
+        .first()
+    )
+    if localization is not None:
+        (localization,) = localization
+    else:
+        raise ValueError(f'No localization with ID {localization_id}')
+
+    instrument = (
+        DBSession()
+        .execute(sa.select(Instrument).where(Instrument.id == instrument_id))
+        .first()
+    )
+    if instrument is not None:
+        (instrument,) = instrument
+    else:
+        raise ValueError(f'No instrument with ID {instrument_id}')
```

Comment:
If you replace `execute` with `scalars` you wouldn't have to unpack the results.